### DISCUSSION TITLE
BareSIP code coverage in GitHub Action

### DISCUSSION THREAD

**Discussion by maximilianfridrich** [baresip/baresip#2548] [Ideas]
I have looked into getting a code coverage report for BareSIP (using the selftest), similar to how it is [done in re](https://github.com/baresip/re/blob/main/.github/workflows/coverage.yml).

Would it be desirable to have a code coverage for BareSIP as well?

It needs to be considered which modules are relevant for the coverage report, because most module are not tested by the BareSIP selftest. One option could be to only get a coverage for the BareSIP core by runing `gcovr -r . -f src -s` which currently results in a coverage of 47.6%.

What do you think? If this is something we want, I can create a PR for it.

**Comment by sreimers** [comment]
Sounds good to me, `src` is a good starting point I think.

**Reply by maximilianfridrich** [reply]
Working implementation here: https://github.com/baresip/baresip/pull/2550

Current coverage: 47.6, minimum coverage: 47.5.
